File: Semestru 4/MPP/lab/.net/lab3/lab3/Networking/protocol/Response.cs
using Model.app.domain;

namespace Networking.rpcprotocol
{
	[Serializable]
	public class Response<ID> {
		public ResponseType? Type { get; set; }
		public Entity<ID>? Entity { get; set; }
		public IEnumerable<Entity<ID>>? Entities { get; set; }
		public string? ErrorMessage { get; set; }

		public override string ToString() =>
			$"Response{{type={Type}, entity={Entity}, entities={Entities}, errorMessage={ErrorMessage}}}";
	}
}

File: Semestru 4/MPP/lab/java/lab3/Network/src/main/java/org/example/protocol/Response.java
package org.example.protocol;

import org.example.domain.Entity;

import java.io.Serializable;
import java.util.List;

public class Response implements Serializable {
    private ResponseType Type;
    private Entity Entity;
    private List<Entity> Entities;
    private String Message;

    public ResponseType getType() {
        return Type;
    }

    public Response setType(ResponseType type) {
        Type = type;
        return this;
    }

    public Entity getEntity() {
        return Entity;
    }

    public Response setEntity(Entity entity) {
        Entity = entity;
        return this;
    }

    public List<Entity> getEntities() {
        return Entities;
    }

    public Response setEntities(List<Entity> entities) {
        Entities = entities;
        return this;
    }

    public String getMessage() {
        return Message;
    }

    public Response setMessage(String message) {
        Message = message;
        return this;
    }

    @Override
    public String toString() {
        return "Response [Type= " + Type + ", Entity= " + Entity + ", Entities= " + Entities + ", Message= " + Message + "]";
    }
}
